integration/spark-common/src/main/scala/org/apache/spark/sql/hive/DistributionUtil.scala

```diff
@@ -89,7 +89,16 @@ object DistributionUtil {
     }
   }
 
-  def getExecutors(sparkContext: SparkContext): Map[String, Seq[String]] = {
+  def assignAndGetExecutors(sparkContext: SparkContext): Map[String, Seq[String]] = {
+    val availableExecutors = getExecutors(sparkContext)
+    if (availableExecutors.isEmpty) {
+      val configuredExecutorInstances = sparkContext.getConf.getInt("spark.executor.instances", 0)
+      sparkContext.requestExecutors(configuredExecutorInstances)
```

Comment:
doesn't it take time time to get the requested executors? Don't we need to wait after this call?

Reply:
Actually ill remove this code..after analysis there are some scenarios that are not possible to handle with dynamicAllocation = true. So instead of this we will not start index server if dynamicAllocation is true.